sdk/inc/azure/core/az_json.h

```diff
@@ -175,6 +175,8 @@ AZ_NODISCARD AZ_INLINE az_json_builder_options az_json_builder_options_default()
   return options;
 }
 
+// TODO: Should total_bytes_written be made public or should az_json_builder_get_json take a
+// destination buffer?
```

Comment:
@JeffreyRichter what are your thoughts on this? With the current design `az_json_builder_get_json(...)` can't return back the complete JSON when the destination could be non-contiguous.

For that, we need to expose the `total_bytes_written` field and/or add an API that lets users get the total JSON into a single buffer.